rita/src/rita_common/tunnel_manager/mod.rs

```diff
@@ -261,18 +263,24 @@ impl TunnelManager {
             .map(|&(ip_address, ref dev)| (ip_address.to_string(), Some(dev.clone())))
             .chain({
                 let mut out = Vec::new();
-                for i in SETTING.get_network().manual_peers.clone() {
-                    out.push((i, None))
+                let settings = SETTING.get_network();
+                let mut iface = None;
+                if settings.default_route.len() > 0 {
+                    let idx = settings
+                        .default_route
+                        .iter()
+                        .position(|s| s == "dev")
+                        .unwrap();
+                    iface = Some(settings.default_route[idx + 1].clone());
+                }
+
+                for i in settings.manual_peers.clone() {
+                    out.push((i, iface.clone()))
```

Comment:
while this gets us the right interface it doesn't actually cause the peer discovery to work. I assume the http request doens't get routed right even though ping -i <wan dev> 1.1.1.1 works regardless of the default route, wonder why that is.

Reply:
None stands for the wan interface (sorry probs not very clear, so I think what you did to get the wan interface name is redundant (but it might be better to always explicitly specify interfaces))

Reply:
@jkilpatr you probably want the `find()` method here instead of `position()`. Also, why the `unwrap`?